### PR TITLE
feat: add flag to run serve as a platformApp

### PR DESCRIPTION
**Pull request by julianiff**

### CODE REVIEW

**Comment by julianiff** [comment]
@marguslant0 and @SamuelAlev i added the optional --appType flag and the read on the manifest to decide on the devServer that will be run.

File: packages/cli/src/index.ts
Comment (SamuelAlev) [inline packages/cli/src/index.ts:1]: Could you add a changeset?

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -78,8 +80,16 @@ cli.command('serve', 'serve the app locally')
     .option('--allowExternal, --allow-external', '[boolean] allow external IPs to access the server', {
         default: false,
     })
+    .option('--appType [appType], --app-type', '[string] specify app type. Overrides manifest values')
```
Comment (marguslant0) [inline packages/cli/src/index.ts:83]: Is there a reason why `[appType]` is in brackets? 🤔 

Should it perhaps be:
`.option('--appType, --app-type, <appType>', ...` instead? (taking inspiration from the `entryPath`)

File: packages/cli/src/index.ts [hunk 2]
```diff
@@ -78,8 +80,16 @@ cli.command('serve', 'serve the app locally')
...
     .action(async (options) => {
-        await createDevelopmentServer(options.entryPath, options.port, options.allowExternal);
+        const manifest = reactiveJson<AppManifest>(join(process.cwd(), 'manifest.json'));
+        const appType = options.appType ?? manifest.appType;
```
Comment (SamuelAlev) [inline packages/cli/src/index.ts:86]:
```suggestion
        const appType = options.appType || manifest.appType;
```
`'' | 'theme' === ''`

File: packages/cli/src/index.ts (hunk 1 above)
Comment (SamuelAlev) [reply packages/cli/src/index.ts:83]: @julianiff Still open?

Comment (julianiff) [reply packages/cli/src/index.ts:83]: ahh brackets means its optional. sorry we cleared it in a 1on1. 
From [cac npm](https://www.npmjs.com/package/cac#brackets): 
When using brackets in command name, angled brackets indicate required command arguments, while square bracket indicate optional arguments.

**Pull request merged**